### ISSUE TITLE
Error when consuming the generated library: Unexpected value '[object Object]' imported by the module 'AppModule'. Please add a @NgModule annotation

### ISSUE THREAD

**Issue by tinesoft** [tinesoft/generator-ngx-library#70]
### Bug Report or Feature Request (mark with an `x`)
```
- [ x] bug report -> please search issues before submitting
- [ ] feature request
```

### OS and Version?

Mac OS Sierra, 10.12.6
### Node / Npm or Yarn Version

node: 8.4.0
npm: 5.3.0
yo: 2.0.0
yarn: 0.27.5
### Targeted Angular version

4.0.0
### Repro steps.

run gulp serve:demo on a freshly generated project
### The log given by the failure.

```
Uncaught Error: Unexpected value '[object Object]' imported by the module 'AppModule'. Please add a @NgModule annotation.
    at syntaxError (compiler.es5.js:1690)
    at compiler.es5.js:15382
    at Array.forEach (<anonymous>)
    at CompileMetadataResolver.webpackJsonp.../../../compiler/@angular/compiler.es5.js.CompileMetadataResolver.getNgModuleMetadata (compiler.es5.js:15365)
    at JitCompiler.webpackJsonp.../../../compiler/@angular/compiler.es5.js.JitCompiler._loadModules (compiler.es5.js:26795)
    at JitCompiler.webpackJsonp.../../../compiler/@angular/compiler.es5.js.JitCompiler._compileModuleAndComponents (compiler.es5.js:26768)
    at JitCompiler.webpackJsonp.../../../compiler/@angular/compiler.es5.js.JitCompiler.compileModuleAsync (compiler.es5.js:26697)
    at PlatformRef_.webpackJsonp.../../../core/@angular/core.es5.js.PlatformRef_._bootstrapModuleWithZone (core.es5.js:4536)
    at PlatformRef_.webpackJsonp.../../../core/@angular/core.es5.js.PlatformRef_.bootstrapModule (core.es5.js:4522)
    at Object.../../../../../src/main.ts (main.ts:11)
```
### Mention any other details that might be useful.

Was working great in `v3.0.0` of the generator

**Comment by tinesoft** [comment]
Hi @gatimus

To sum up, there is a limitation when consuming a **linked** library that has `peerDependencies`, as explained [here](http://codetunnel.io/you-can-finally-npm-link-packages-that-contain-peer-dependencies/). Adding `--preserve-symlinks` to the launched command, solves the issue.
I have updated the [README.md](https://github.com/tinesoft/generator-ngx-library#development) section about that.

I've just released  [v3.1.1](https://github.com/tinesoft/generator-ngx-library/releases/tag/v3.1.1) with the fix for that issue.

Please update and let me know if it is working fine for you.

Cheers,

**Comment by gatimus** [comment]
@tinesoft, yes working now, thank you. I also learned you have to make sure to do `ng serve --preserve-symlinks` not ` npm start --preserve-symlinks` from the consuming app.
